### PR TITLE
Try to reduce amount of time on the asmjs builder

### PR DESCRIPTION
**Pull request by alexcrichton**
This PR has two commits for two separate strategies:

* First it disables optimizations for all tests, hopefully saving time by not optimizing the test code. This caused a number of run-pass tests to fail which are switched to being ignored here.
* Next it disables a number of test suites which aren't asm.js specific and already run elsewhere

cc #48826

### CODE REVIEW

File: src/ci/docker/asmjs/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,11 @@ ENV EM_CONFIG=/emsdk-portable/.emscripten
 
 ENV TARGETS=asmjs-unknown-emscripten
 
-ENV RUST_CONFIGURE_ARGS --enable-emscripten
+ENV RUST_CONFIGURE_ARGS --enable-emscripten --disable-optimize-tests
 
-ENV SCRIPT python2.7 ../x.py test --target $TARGETS
+ENV SCRIPT python2.7 ../x.py test --target $TARGETS \
+  src/test/run-pass \
+  src/test/run-fail \
```
Comment (kennytm) [inline src/ci/docker/asmjs/Dockerfile]: I think we should still enable `run-pass-fulldeps`, `run-fail-fulldeps` and `run-make`.

Comment (alexcrichton) [reply src/ci/docker/asmjs/Dockerfile]: Oh currently I don't believe we run any of those tests, right?

Comment (kennytm) [reply src/ci/docker/asmjs/Dockerfile]: Oh let me check...

**Pull request merged**